packages/flutter/lib/src/services/message_codecs.dart

```diff
@@ -293,12 +302,36 @@ class StandardMessageCodec implements MessageCodec<dynamic> {
   /// This method may be called recursively to serialize container values.
   ///
   /// Type discriminators 0 through 127 inclusive are reserved for use by the
-  /// base class.
+  /// base class, as follows:
+  ///
+  ///  * null = 0
+  ///  * true = 1
+  ///  * false = 2
+  ///  * 32 bit integer = 3
+  ///  * 64 bit integer = 4
+  ///  * larger integers = 5 (see below)
+  ///  * 64 bit floating-point number = 6
+  ///  * String = 7
+  ///  * Uint8List = 8
+  ///  * Int32List = 9
+  ///  * Int64List = 10
+  ///  * Float64List = 11
+  ///  * List = 12
+  ///  * Map = 13
+  ///  * Reserved for future expansion: 14..127
   ///
   /// The codec can be extended by overriding this method, calling super
   /// for values that the extension does not handle. Type discriminators
   /// used by extensions must be greater than or equal to 128 in order to avoid
   /// clashes with any later extensions to the base class.
+  ///
+  /// The "larger integers" type, 5, is never used by [writeValue]. A subclass
+  /// could represent big integers from another package using that type. The
```

Comment:
Do you have some historical context as for whether we can just deprecate it? (the comment on the decoder for this mentions we are planning to deprecate it)

I just find it a little weird to have us specify the format, keep a decoder implementation but say if you want to encode subclass and implement exactly this format...

Reply:
Coming back here after seeing the readValueOfType change which is backwards incompatible anyways... why not just drop support for this type?

Reply:
We said we'd remove it a year ago. We didn't. We missed the boat on deprecation. This is just an attempt to rescue a clean API out of the ashes that remain.